File: solutions/java/log-levels/1/src/main/java/LogLevels.java


public class LogLevels {

    enum Level {
        INFO,
        WARNING,
        ERROR
    }
    
    public static String message(String logLine) {
        return logLine.split(":")[1].trim();
    }

    public static String logLevel(String logLine) {
        return logLine.split(":")[0].replace("[", "").replace("]", "").toLowerCase();
    }

    public static String reformat(String logLine) {
        String msg = message(logLine);
        String level = logLevel(logLine);
        return msg + " (" + level + ")";
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
using System.Text.RegularExpressions;

static partial class LogLine
{

    [GeneratedRegex("\\[(?<logLevel>\\w+)\\]")]
    private static partial Regex LogLevelRegex();

    public static string Message(string logLine)
    {
        return logLine.Split("]: ").Last().Trim();
    }

    public static string LogLevel(string logLine)
    {
        var regex = LogLevelRegex();

        return regex.Match(logLine).Groups["logLevel"].Captures.First().ToString().ToLower();
    }

    public static string Reformat(string logLine)
    {
        var logLevel = LogLevel(logLine);
        var message = Message(logLine);
        
        return $"{message} ({logLevel})";
    }

}